src/AzureClient/Magic/TargetMagic.cs

```diff
@@ -3,95 +3,112 @@
+                    in an Azure Quantum workspace.
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="TargetMagic"/> class.
-        /// </summary>
-        /// <param name="azureClient">
-        /// The <see cref="IAzureClient"/> object to use for Azure functionality.
-        /// </param>
-        /// <param name="logger">Logger instance for messages.</param>
-        public TargetMagic(IAzureClient azureClient, ILogger<TargetMagic> logger)
-            : base(
-                azureClient,
-                "azure.target",
-                new Microsoft.Jupyter.Core.Documentation
-                {
-                    Summary = "Sets or displays the active execution target for Q# job submission in an Azure Quantum workspace.",
-                    Description = $@"
-                        This magic command allows for specifying or displaying the execution target for Q# job submission
-                        in an Azure Quantum workspace.
+                    The Azure Quantum workspace must have been previously initialized
+                    using the [`%azure.connect` magic command]({KnownUris.ReferenceForMagicCommand("azure.connect")})
+                    magic command. The specified execution target must be available in the workspace and support execution of Q# programs.
+
+                    #### Optional parameters
 
-                        The Azure Quantum workspace must have been previously initialized
-                        using the [`%azure.connect` magic command]({KnownUris.ReferenceForMagicCommand("azure.connect")})
-                        magic command. The specified execution target must be available in the workspace and support execution of Q# programs.
+                    - The target ID to set as the active execution target for Q# job submission. If not specified,
+                    the currently active execution target is displayed.
 
-                        #### Optional parameters
+                    #### Possible errors
 
-                        - The target ID to set as the active execution target for Q# job submission. If not specified,
-                        the currently active execution target is displayed.
+                    - {AzureClientError.NotConnected.ToMarkdown()}
+                    - {AzureClientError.InvalidTarget.ToMarkdown()}
+                    - {AzureClientError.NoTarget.ToMarkdown()}
 
-                        #### Possible errors
+                    #### Target capabilities
 
-                        - {AzureClientError.NotConnected.ToMarkdown()}
-                        - {AzureClientError.InvalidTarget.ToMarkdown()}
-                        - {AzureClientError.NoTarget.ToMarkdown()}
+                    When setting a target, the target capability is set to
+                    the maximum capability level supported by the given
```

Comment:
FYI: the fragment "_maximum_ capability level supported" seems somewhat vague to me and somewhat interpretable in the opposite ways (I'm not very familiar with target capabilities, have mostly intuitive understanding). I would prefer instead of the word "maximum" to see something more specific, like "earliest" or "latest" or "most-demanding" or "least-demanding", etc.